### PR TITLE
Fix siret and siren generation for fr_FR locale company information

### PR DESCRIPTION
**Pull request by daamien**
Hi !

This is a new attempt to fix the French SIRET/SIREN generation.

The correction was first submitted in MR #1465 in 2021 but never merged

I simply did a rebase from the master branch and I added basic tests

### CODE REVIEW

File: faker/providers/company/fr_FR/__init__.py [hunk 1]
```diff
@@ -121,20 +122,28 @@ def _is_catch_phrase_valid(self, catch_phrase: str) -> bool:
 
     def siren(self) -> str:
         """
-        Generates a siren number (9 digits).
+        Generates a siren number (9 digits). Formated as '### ### ###'.
```
Comment (stefan6419846) [inline faker/providers/company/fr_FR/__init__.py:125]: Should probably read `Formatted` instead.

File: tests/providers/test_company.py [hunk 2]
```diff
@@ -60,6 +62,22 @@ def test_company_business_id(self, faker, num_samples):
...
 
 
+class TestFrFr:
+    """Test fr_FR company provider methods"""
+
+    def test_siren(self, faker, num_samples):
+        for _ in range(num_samples):
+            siren = faker.siren()
+            assert len(siren) == 11
+            assert luhn_checksum(siren.replace(' ','')) == 0
```
Comment (stefan6419846) [inline tests/providers/test_company.py:72]: These should probably read `siren.replace(' ', '')` instead.

**Review by fcurella — CHANGES_REQUESTED** [review]
can you run `make lint` to fix the formatting issues?

**Comment by daamien** [comment]
> can you run `make lint` to fix the formatting issues?

done !

**Pull request merged**